payjoin/src/receive/mod.rs

```diff
@@ -791,7 +807,25 @@ impl PayjoinProposal {
             self.payjoin_psbt.inputs[i].final_script_sig = None;
             self.payjoin_psbt.inputs[i].final_script_witness = None;
         }
-        Ok(self.payjoin_psbt)
+        Ok(PayjoinProposal {
+            payjoin_psbt: self.payjoin_psbt,
+            owned_vouts: self.owned_vouts,
+            params: self.params,
+        })
+    }
+
+    pub fn finalize_proposal(
+        mut self,
+        wallet_process_psbt: impl Fn(&Psbt) -> Result<Psbt, Error>,
+        min_feerate_sat_per_vb: Option<FeeRate>,
+    ) -> Result<PayjoinProposal, RequestError> {
+        let psbt = self.apply_fee(min_feerate_sat_per_vb)?;
+        let psbt = wallet_process_psbt(psbt).map_err(|_| {
```

Comment:
better to handle the specific error instead of discarding it by returning `Error` from finalize_proposal instead of `RequestError`. Then we don't have to convert after `prepare_psbt` either